include/pybind11/gil.h

```diff
@@ -21,7 +21,7 @@ PyThreadState *get_thread_state_unchecked();
 
 PYBIND11_NAMESPACE_END(detail)
 
-#if defined(WITH_THREAD) && !defined(PYPY_VERSION)
+#if defined(WITH_THREAD) && !defined(PYPY_VERSION) && !defined(PYBIND11_SIMPLE_GIL)
```

Comment:
Slight preference to formalize this more:

In detail/common.h (untested):
```
#if defined(PYPY_VERSION) && !defined(PYBIND11_SIMPLE_GIL_MANAGEMENT)
#    define PYBIND11_SIMPLE_GIL_MANAGEMENT
#endif
```

Then simply use `!defined(PYBIND11_SIMPLE_GIL_MANAGEMENT)` here and below.

That way it's easier to play with the condition (only one place). We could even use it in internals.h to `#ifdef` out the `gil_scoped_acquire_local` block (tiny code size reduction probably; and won't get missed when the non-simple code is removed).